File: var/spack/repos/builtin/packages/mpi/test/mpi_hello.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char** argv) {
  MPI_Init(&argc, &argv);

  int rank;
  int num_ranks;
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
  MPI_Comm_size(MPI_COMM_WORLD, &num_ranks);

  printf("Hello world! From rank %d of %d\n", rank, num_ranks);

  MPI_Finalize();
  return(0);
}

File: var/spack/repos/builtin/packages/mpi/test/mpi_hello.f
c Fortran example
      program hello
      include 'mpif.h'
      integer rank, num_ranks, err_flag

      call MPI_INIT(err_flag)
      call MPI_COMM_SIZE(MPI_COMM_WORLD, num_ranks, err_flag)
      call MPI_COMM_RANK(MPI_COMM_WORLD, rank, err_flag)
      print*, 'Hello world! From rank', rank, 'of ', num_ranks
      call MPI_FINALIZE(err_flag)
      end
